test/Microsoft.Azure.Devices.Edge.Test/SetupFixture.cs

```diff
@@ -62,11 +65,30 @@ public async Task Setup()
                             Context.Current.PackagePath,
                             Context.Current.Proxy,
                             token);
-                        await this.daemon.WaitForStatusAsync(EdgeDaemonStatus.Running, token);
 
-                        var agent = new EdgeAgent(this.device.Id, this.iotHub);
-                        await agent.WaitForStatusAsync(EdgeModuleStatus.Running, token);
-                        await agent.PingAsync(token);
+                        try
+                        {
+                            await this.daemon.WaitForStatusAsync(EdgeDaemonStatus.Running, token);
+
+                            var agent = new EdgeAgent(this.device.Id, this.iotHub);
+                            await agent.WaitForStatusAsync(EdgeModuleStatus.Running, token);
+                            await agent.PingAsync(token);
+                        }
+
+                        // ReSharper disable once RedundantCatchClause
+                        catch
+                        {
+                            throw;
```

Comment:
Why do you need this?

Reply:
The [try-finally docs](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/try-finally) say that the finally block isn't executed in certain cases, and adding a catch is the workaround.